src/network/network_access_manager.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "local_socket_reply.h"
+
+#include <multipass/network_access_manager.h>
+
+namespace mp = multipass;
+
+mp::NetworkAccessManager::NetworkAccessManager(QObject* parent) : QNetworkAccessManager(parent)
+{
+}
+
+QNetworkReply* mp::NetworkAccessManager::createRequest(QNetworkAccessManager::Operation operation,
+                                                       const QNetworkRequest& orig_request, QIODevice* device)
+{
+    auto scheme = orig_request.url().scheme();
+
+    // To support http requests over Unix sockets, the initial URL needs to be in the form of:
+    // unix:///path/to/unix_socket@path/in/server (or 'local' instead of 'unix')
+    //
+    // For example, to get the general LXD configuration when LXD is installed as a snap:
+    // unix:////var/snap/lxd/common/lxd/unix.socket@1.0
```

Comment:
Yeah, that works - if anyone actually wants '@' in their URL, they can always encode it, right?

Reply:
It would be that someone has an `@` in their path to the local socket.  I'm not sure what the chances of that happening, particularly pertaining to the LXD socket since that is really all we care about now.  Since this is an internal way of parsing local sockets and REST paths, then we can always change it later and it should still work.